File: app/src/main/java/com/gmail/fattazzo/formula1world/ergast/json/parser/FieldExclusionStrategy.java
package com.gmail.fattazzo.formula1world.ergast.json.parser;

import com.google.gson.ExclusionStrategy;
import com.google.gson.FieldAttributes;

public class FieldExclusionStrategy implements ExclusionStrategy {
 
   public boolean shouldSkipField(FieldAttributes f) {
      return f.getAnnotation(ExcludeFieldFromJson.class) != null;
   }
 
   public boolean shouldSkipClass(Class<?> clazz) {
      return false;
   }
 
}

File: app/src/main/kotlin/com/gmail/fattazzo/formula1world/ergast/json/parser/FieldExclusionStrategy.kt
package com.gmail.fattazzo.formula1world.ergast.json.parser

import com.google.gson.ExclusionStrategy
import com.google.gson.FieldAttributes

class FieldExclusionStrategy : ExclusionStrategy {

    override fun shouldSkipField(f: FieldAttributes): Boolean {
        return f.getAnnotation(ExcludeFieldFromJson::class.java) != null
    }

    override fun shouldSkipClass(clazz: Class<*>): Boolean {
        return false
    }

}